manifests/init.pp

```diff
@@ -107,13 +107,13 @@
   Optional[String] $fastcgi_cache_path                       = undef,
   Optional[String] $fastcgi_cache_use_stale                  = undef,
   Enum['on', 'off'] $gzip                                    = 'off',
-  $gzip_buffers                                              = undef,
-  $gzip_comp_level                                           = 1,
-  $gzip_disable                                              = 'msie6',
-  $gzip_min_length                                           = 20,
-  $gzip_http_version                                         = 1.1,
+  Optional[String] $gzip_buffers                             = undef,
+  Integer $gzip_comp_level                                   = 1,
+  String $gzip_disable                                       = 'msie6',
+  Integer $gzip_min_length                                   = 20,
+  Float $gzip_http_version                                   = 1.1,
```

Comment:
I know the default is `1.1`, but this doesn't look right to me.  Version 'numbers' should be strings, not floats (and in this case possibly an `Enum`)

Reply:
Maybe
```suggestion
  Enum['1.0','1.1'] $gzip_http_version                       = '1.1',
```

Reply:
Possibly.  I'm guessing for very few users though.  I can't imagine many people setting this to `1.0` (and as a float).

I suppose you _could_ make it a `Variant[Float,Enum['1.0','1.1']]` and then in the code issue some deprecation warning `unless $gzip_http_version =~ String`, but _that_ is a lot of maintenance work for the benefit that, (I'm _assuming_), will be low in this case.

Reply:
I'm fine with anything.  My goal is to enable the parameter types puppet-lint plugin. What about using `Variant[String,Float]` and raise another PR as breaking change that switches to the enum and we keep that open until the next major release? I can follow the same pattern for the other odd types

Reply:
Personally, I wouldn't bother with the plugin if it means creating bad data types though.  IMO, the default being a float` here is just a bug.